### PR TITLE
Patch extra files needed for views deployment

### PR DESCRIPTION
**Pull request by zhan-airplane**
## Description
Still have a bit of cleanup to do, but going to put up this PR first - basically the equivalent of the CLI dev change but for deploy. Needs updated PRs in both `airport` and `cli` that I'm going to put up soon

## Test plan
Ran views init and views deploy using CLI (PR incoming) and verified that it worked. 

---
cc @leeweisberger @stepzhou

### CODE REVIEW

File: pkg/build/view.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+	}
+
+	// TODO: patch package.json correctly.
+	return applyTemplate(heredoc.Doc(`
+		FROM {{.Base}} as builder
+		WORKDIR /airplane
+
+		COPY package*.json yarn.* /airplane/
+		RUN [ -f package.json ] || { echo "{}" > package.json; }
+		RUN yarn add vite @vitejs/plugin-react react react-dom @airplane/views
```
Comment (zhan-airplane) [inline pkg/build/view.go:76]: Going to have to pick specific minimum versions for these probably

File: pkg/build/view.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+}
+
+// view creates a dockerfile for an view.
+func view(root string, options KindOptions) (string, error) {
+	// Assert that the entrypoint file exists:
+	entrypoint, _ := options["entrypoint"].(string)
+	if entrypoint == "" {
+		return "", errors.New("expected an entrypoint")
+	}
+	/*if err := fsx.AssertExistsAll(filepath.Join(root, entrypoint)); err != nil {
```
Comment (rose-huang) [inline pkg/build/view.go:22]: delete this if we're not using it?

File: pkg/build/views/index.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
```
Comment (rose-huang) [inline pkg/build/views/index.html:1]: can we use the same file for dev and deploy for this, the main.tsx file, and the vite config? the dev one is in cli right? also feel free to change this in another PR

File: pkg/examples/typescript/yarnworkspaces/yarn.lock [hunk 4]
```diff
@@ -1,54 +0,0 @@
-# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (rose-huang) [inline pkg/examples/typescript/yarnworkspaces/yarn.lock:1]: was deleting this intentional?

**Review by rose-huang — APPROVED** [review]

File: pkg/build/view.go (hunk 2 above)
Comment (rose-huang) [reply pkg/build/view.go:22]: also are we using root for anything?

File: pkg/build/view.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+import (
+	_ "embed"
+	"strings"
+
+	"github.com/MakeNowJust/heredoc"
+	"github.com/pkg/errors"
+)
+
+func DoView(root string, options KindOptions) (string, error) {
```
Comment (rose-huang) [inline pkg/build/view.go:11]: what is this used for?

Comment (zhan-airplane) [reply pkg/build/view.go:11]: Oops was using this to test this function, will delete

File: pkg/build/views/index.html (hunk 3 above)
Comment (zhan-airplane) [reply pkg/build/views/index.html:1]: Yup I'm planning to shift a bunch of code from cli to lib after this merges

**Pull request merged**